### PR TITLE
Solve crash problem in Safari with ui-options

### PR DESCRIPTION
**Pull request by xmlilley**
See: #38

### CODE REVIEW

File: src/mask.js [hunk 1]
```diff
@@ -110,9 +110,7 @@ angular.module('ui.mask', [])
                                             if (Object.prototype.hasOwnProperty.call(original, i)) {
                                                 if (current[i] === undefined) {
                                                     current[i] = angular.copy(original[i]);
-                                                } else {
```
Comment (NickTomlin) [inline src/mask.js]: I'm not sure if simply removing the extend here is the best option? Can we guard against this specific case in another way?

**Comment by xmlilley** [comment]
Not at all dull. And I realize I'm failing a test now. Need to setup a full local build to deal with that... will be back in a bit...

**Comment by xmlilley** [comment]
OK, let's see if we can step through this. When you hit line 114, you're dealing with the individual properties of either `uiMaskConfig` or of `ui-options`. 

Some of those props are themselves objects, for which `angular.extend()` is a great solution to augmenting what came in on `ui-options` with anything it needs from `uiMaskConfig`. But others, like `clearOnBlur` are a primitive (or an array, like `eventsToHandle`).

So, we wind up calling `angular.extend(false, true)` when we get to the parsing and extension of `clearOnBlur`. In Chrome, if you run `var test = angular.extend(false, true);`, it's OK. `test` is `false`. But in Safari, `var test = angular.extend(false, true);` throws: `TypeError: Requested keys of a value that is not an object.`

So, I now understand the use-case a bit better, and realize we do indeed need to do `angular.extend()` for anything that's an object. So, I'd suggest something more along the lines of: 

```js
linkOptions = (function(original, current) {
    for (var i in original) {
        if (Object.prototype.hasOwnProperty.call(original, i)) {
            if (current[i] === undefined) {
                current[i] = angular.copy(original[i]);
            } else {
                if (angular.isObject(current[i])) {
                    angular.extend(current[i], original[i]);
                }
            }
        }
    }
    return current;
})(options, linkOptions[0]);
```

With that, the tests seem to pass, and I can successfully use `clearOnBlur: false`.

**Comment by xmlilley** [comment]
(Note: if and when you ever release a version that supports only angular 1.4, we can use the new `angular.merge()`, which solves for all of this with one line of code.)

**Comment by PowerKiKi** [comment]
LGTM, @NickTomlin any opinion on that new version ?

**Comment by NickTomlin** [comment]
@xmlilley thanks for the details. This looks great :+1:

**Comment by xmlilley** [comment]
Sorry I've done this iteratively, over multiple commits. If you'd like me to close and resubmit it squashed, let me know.

**Comment by NickTomlin** [comment]
@xmlilley Feel free to squash, but I think this is fine as-is. I don't actually have the ability to merge this, but PowerKiKi should get to this soon. Thanks!

**Pull request merged**